src/trace_data.rs

```diff
@@ -130,3 +130,34 @@ pub enum CallType {
     Call = 0,
     Delegate = 1,
 }
+
+impl ExecutionResources {
```

Comment:
I added it here as an util, idk if it should be this way but it does not hurt much atm.